### PR TITLE
Normalise email address value before adding to index table

### PR DESCRIPTION
**Pull request by eshryane**

### CODE REVIEW

**Comment by eshryane** [comment]
Integration tests passed 8ddb94b

File: whois-commons/src/main/java/net/ripe/db/whois/common/dao/jdbc/index/IndexWithEmailAddressValueAndType.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    @Override
+    public int addToIndex(JdbcTemplate jdbcTemplate, RpslObjectInfo objectInfo, RpslObject object, String value) {
+        return super.addToIndex(jdbcTemplate, objectInfo, object, normaliseEmailAddress(value));
+    }
+
+    private String normaliseEmailAddress(final String emailAddress) {
+        try {
+            return EMAIL_PARSER.parse(emailAddress).getAddress();
```
Comment (maggarwal13) [inline whois-commons/src/main/java/net/ripe/db/whois/common/dao/jdbc/index/IndexWithEmailAddressValueAndType.java:29]: this function is  duplicated with  indexWithEmailAddressValue?

File: whois-commons/src/main/java/net/ripe/db/whois/common/dao/jdbc/index/IndexWithEmailAddressValueAndType.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    private static final Logger LOGGER = LoggerFactory.getLogger(IndexWithEmailAddressValueAndType.class);
+
+    private static final AttributeParser.EmailParser EMAIL_PARSER = new AttributeParser.EmailParser();
+
+    public IndexWithEmailAddressValueAndType(final AttributeType attributeType, final String lookupTableName, final String lookupColumnName) {
+        super(attributeType, lookupTableName, lookupColumnName);
+    }
+
+    @Override
+    public int addToIndex(JdbcTemplate jdbcTemplate, RpslObjectInfo objectInfo, RpslObject object, String value) {
```
Comment (maggarwal13) [inline whois-commons/src/main/java/net/ripe/db/whois/common/dao/jdbc/index/IndexWithEmailAddressValueAndType.java:23]: shoudln't we make sure that this function is called for email type only?

File: whois-commons/src/main/java/net/ripe/db/whois/common/dao/jdbc/index/IndexWithEmailAddressValueAndType.java (hunk 1 above)
Comment (eshryane) [reply whois-commons/src/main/java/net/ripe/db/whois/common/dao/jdbc/index/IndexWithEmailAddressValueAndType.java:29]: There's no easy way to share this method as the classes extend different classes (the hierarchy is different)

File: whois-commons/src/main/java/net/ripe/db/whois/common/dao/jdbc/index/IndexWithEmailAddressValueAndType.java (hunk 2 above)
Comment (eshryane) [reply whois-commons/src/main/java/net/ripe/db/whois/common/dao/jdbc/index/IndexWithEmailAddressValueAndType.java:23]: That's done in `IndexStrategies.java` you define which strategy to use for different attribute types. This class is only added to attributes that contain an email address.

File: whois-commons/src/main/java/net/ripe/db/whois/common/dao/jdbc/index/IndexWithEmailAddressValueAndType.java (hunk 1 above)
Comment (eshryane) [reply whois-commons/src/main/java/net/ripe/db/whois/common/dao/jdbc/index/IndexWithEmailAddressValueAndType.java:29]: We have to extend `IndexWithValue` and `IndexWithValueAndType` with separate classes

File: whois-commons/src/main/java/net/ripe/db/whois/common/dao/jdbc/index/IndexWithEmailAddressValueAndType.java (hunk 2 above)
Comment (maggarwal13) [reply whois-commons/src/main/java/net/ripe/db/whois/common/dao/jdbc/index/IndexWithEmailAddressValueAndType.java:23]: Normalise email address function can be static to make sure both the functions are doing same thing and code is not duplicated. May be shift it to util class?

Comment (eshryane) [reply whois-commons/src/main/java/net/ripe/db/whois/common/dao/jdbc/index/IndexWithEmailAddressValueAndType.java:23]: Good idea. I'll make it static but I'll leave it in one of the Index classes and call it from the other one

Comment (eshryane) [reply whois-commons/src/main/java/net/ripe/db/whois/common/dao/jdbc/index/IndexWithEmailAddressValueAndType.java:15]: Updated code changes

File: whois-commons/src/main/java/net/ripe/db/whois/common/dao/jdbc/index/IndexWithEmailAddressValueAndType.java (hunk 1 above)
Comment (eshryane) [reply whois-commons/src/main/java/net/ripe/db/whois/common/dao/jdbc/index/IndexWithEmailAddressValueAndType.java]: Updated code changes

**Pull request merged**